### PR TITLE
[SYCL] Cleanup/refactor device_code_tests

### PR DESCRIPTION
**Pull request by ianayl**
This PR:
- Changes more lit tests in sycl/test/check_device_code to use SYCL_EXTERNAL functions in order to remove redundant host-side boilerplate code, and
- Moves tests not verifying device code outside of the check_device_code folder.

### CODE REVIEW

**Comment by ianayl** [comment]
This is a re-do of https://github.com/intel/llvm/pull/14063 after my mess-up: All the commits in this PR should have been approved in #14063 already.

File: sycl/test/check_device_code/fpga_ihs_float.cpp [hunk 1]
```diff
@@ -19,431 +19,387 @@ constexpr bool ToSign = true;
...
 
 template <int EA, int MA, int Eout, int Mout>
-void ap_float_cast() {
+SYCL_EXTERNAL auto ap_float_cast() {
   sycl::detail::ap_int<1 + EA + MA> A;
-  sycl::detail::ap_int<1 + Eout + Mout> float_cast_res =
-      __spirv_ArbitraryFloatCastINTEL<1 + EA + MA, 1 + Eout + Mout>(
-          A, MA, Mout, Subnorm, RndMode, RndAcc);
-  // CHECK: call spir_func i40 @_Z{{[0-9]+}}__spirv_ArbitraryFloatCastINTEL{{.*}}(i40 {{[%a-z0-9.]+}}, i32 28, i32 30, i32 0, i32 2, i32 1)
+  return __spirv_ArbitraryFloatCastINTEL<1 + EA + MA, 1 + Eout + Mout>(
```
Comment (aelovikov-intel) [inline sycl/test/check_device_code/fpga_ihs_float.cpp:24]: I' d move local variables to the function arguments.

**Review by aelovikov-intel — APPROVED** [review]

File: sycl/test/check_device_code/fpga_ihs_float.cpp (hunk 1 above)
Comment (ianayl) [reply sycl/test/check_device_code/fpga_ihs_float.cpp:24]: Fixed!

**Review by aelovikov-intel — APPROVED** [review]

**Comment by ianayl** [comment]
@sergey-semenov Please excuse the ping Sergey, do I need your approval to merge?

**Comment by sergey-semenov** [comment]
@ianayl You don't, Andrei's approval covers the intel/llvm-reviewers-runtime review requirement. This will get merged by one of the gatekeepers.

**Pull request merged**